### PR TITLE
[IMP] mrp: add hook to prepare procurement values on manufacture creation

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
We need  a hook to be able to personalize the procurement values which will be used on procurement group creation when creating manufacturing orders.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
@sle-odoo any inconvenience with this? Thanks for your time.

**Comment by LoisRForgeFlow** [comment]
@sle-odoo I wanted to say hi also here :smile:

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -495,6 +495,10 @@ def write(self, vals):
                 production.move_raw_ids.filtered(lambda m: m.state == 'draft')._action_confirm()
         return res
 
+    @api.model
+    def prepare_procurement_group(self, values):
+        return {'name': values['name']}
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_production.py:500]: should not be public + this kind of helper should be at the end of the file + _prepare_procurement_group_vals + self.ensure_one()

- Private attributes (_name, _description, _inherit, …)
- Default method and _default_get
- Field declarations
- Compute, inverse and search methods in the same order as field declaration
- Selection method (methods used to return computed values for selection fields)
- Constrains methods (@api.constrains) and onchange methods (@api.onchange)
- CRUD methods (ORM overrides)
- Action methods
- And finally, other business methods.

**Comment by AaronHForgeFlow** [comment]
Thank you for considering this PR @sle-odoo 

I have updated this according to your comments. Just let me know if any other change is required.

Thanks for your time.

**Comment by sle-odoo** [comment]
robodoo r+

**Comment by AaronHForgeFlow** [comment]
I had to remove the self.ensure_one() because the prepare method is @api.model and fails

**Comment by AaronHForgeFlow** [comment]
Hi @sle-odoo  I just rebased this PR and checks passed.

just let me know of anything else is required to get this merged. Thank you!

**Comment by sle-odoo** [comment]
robodoo r+

**Pull request closed without merging**